tools/cpp/windows_cc_configure.bzl

```diff
@@ -269,28 +274,44 @@ def setup_vc_env_vars(repository_ctx, vc_path):
         env_map[key] = escape_string(value.replace("\\", "\\\\"))
     return env_map
 
+def _get_latest_subversion(repository_ctx, vc_path):
+    """Get the latest subversion of a VS 2017/2019 installation.
+
+    For VS 2017 & 2019, there could be multiple versions of VC build tools.
+    The directories are like:
+      <vc_path>\\Tools\\MSVC\\14.10.24930\\bin\\HostX64\\x64
+      <vc_path>\\Tools\\MSVC\\14.16.27023\\bin\\HostX64\\x64
+    This function should return 14.16.27023 in this case."""
+    versions = [path.basename for path in repository_ctx.path(vc_path + "\\Tools\\MSVC").readdir()]
+    if len(versions) < 1:
+        _auto_configure_warning_maybe(repository_ctx, "Cannot find any VC installation under BAZEL_VC(%s)" % vc_path)
+        return None
+
+    versions = sorted(versions)
```

Comment:
This will do textual sorting, which may sort `10` before `9`.
Could you split the version strings on `.`, parse the components as int, and sort by components?

Reply:
Good suggestion, done!

Reply:
I also tested this by having a fake version with `9.17.27023`, without the latest change, Bazel will wrongly choose this one.